django/utils/translation/__init__.py

```diff
@@ -107,6 +107,12 @@ def lazy_number(func, resultclass, number=None, **kwargs):
         proxy = lazy(func, resultclass)(**kwargs)
     else:
         class NumberAwareString(resultclass):
+            def __bool__(self):
+                return bool(kwargs['singular'])
```

Comment:
Thanks -- is there ever a case where singular is empty but not plural? I thought `kwargs['singular'] or kwargs['plural']` could guard against that if it makes sense.